KeywordSearch/src/org/sleuthkit/autopsy/keywordsearch/SqliteTextExtractor.java

```diff
@@ -91,267 +84,251 @@ boolean isSupported(Content file, String detectedFormat) {
-         *
-         * @return Current table contents or null meaning there are not more
-         *         tables to process
-         */
-        private String getNextTable() {
-            if (tableIterator.hasNext()) {
-                return getTableAsString(tableIterator.next());
-            } else {
-                return null;
+            //Keep grabbing table names from the queue and reading them until
+            //our buffer is full.
+            while (bufIndex != len) {
+                if (Objects.isNull(currentTableName) || reader.isFinished()) {
+                    if (tableNames.hasNext()) {
+                        currentTableName = tableNames.next();
+                        try {
+                            totalColumns = reader.getColumnCount(currentTableName);
+                            reader.read(currentTableName, () -> bufIndex == len);
+                        } catch (SQLiteTableReaderException ex) {
+                            logger.log(Level.WARNING, String.format(
+                                "Error attempting to read file table: [%s]" //NON-NLS
+                                + " for file: [%s] (id=%d).", currentTableName, //NON-NLS
+                                file.getName(), file.getId()), ex.getMessage());
+                        }
+                    } else {
+                        if (bufIndex == off) {
+                            return -1;
+                        }
+                        return bufIndex;
+                    }
+                } else {
+                    try {
+                        reader.read(currentTableName, () -> bufIndex == len);
+                    } catch (SQLiteTableReaderException ex) {
+                        logger.log(Level.WARNING, String.format(
+                                "Error attempting to read file table: [%s]" //NON-NLS
+                                + " for file: [%s] (id=%d).", currentTableName, //NON-NLS
+                                file.getName(), file.getId()), ex.getMessage());
+                    }
+                }
             }
+
+            return bufIndex;
         }
 
-        /**
-         * Close the underlying connection to the database.
-         *
-         * @throws IOException Not applicable, we can just catch the
-         *                     SQLException
-         */
         @Override
         public void close() throws IOException {
             try {
-                connection.close();
-            } catch (SQLException ex) {
-                //Non-essential exception, user has no need for the connection 
-                //object at this stage so closing details are not important
-                logger.log(Level.WARNING, "Could not close JDBC connection", ex);
+                reader.close();
+            } catch (SQLiteTableReaderException ex) {
+                //Done reading, but couldn't close the resources. Nothing we can 
+                //do as a recovery.
```

Comment:
It is fine to not propagate the exception here, but I would prefer that we log it.